source/plugins/go/input/lib/applicationinsights.go

```diff
@@ -0,0 +1,339 @@
+package lib
+
+import (
+	"context"
+	"encoding/base64"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/docker/docker/client"
+	"github.com/microsoft/ApplicationInsights-Go/appinsights"
+)
+
+const (
+	heartBeat       = "HeartBeatEvent"
+	exception       = "ExceptionEvent"
+	acsClusterType  = "ACS"
+	aksClusterType  = "AKS"
+	envAcsResource  = "ACS_RESOURCE_NAME"
+	envAksRegion    = "AKS_REGION"
+	envAgentVersion = "AGENT_VERSION"
+	envAppInsights  = "APPLICATIONINSIGHTS_AUTH"
+	envAppEndpoint  = "APPLICATIONINSIGHTS_ENDPOINT"
+	envController   = "CONTROLLER_TYPE"
+	envContainerRT  = "CONTAINER_RUNTIME"
+	envAADMSIAuth   = "AAD_MSI_AUTH_MODE"
+	envAddonResizer = "RS_ADDON-RESIZER_VPA_ENABLED"
+)
+
+var (
+	isWindows        bool
+	hostName         string
+	customProperties map[string]string
+	telemetryClient  appinsights.TelemetryClient
+	proxyEndpoint    string
+	aiLogger         *log.Logger
+)
+
+func init() {
+	// Check if the OS is Windows
+	osType := os.Getenv("OS_TYPE")
+	hostName = os.Getenv("HOSTNAME")
+	if strings.TrimSpace(strings.ToLower(osType)) == "windows" {
+		isWindows = true
+	} else {
+		isWindows = false
+	}
+	var logPath string
+	if isWindows {
+		logPath = "/etc/amalogswindows/appinsights_error.log"
+	} else {
+		logPath = "/var/opt/microsoft/docker-cimprov/log/appinsights_error.log"
+	}
+
+	aiLogger = CreateLogger(logPath)
+	// Initialize customProperties
+	customProperties = make(map[string]string)
+	if !isIgnoreProxySettings() {
+		if isWindows {
+			proxyEndpoint = os.Getenv("PROXY")
+
+		} else {
+			proxyEndpoint = getProxyEndpoint()
+		}
+	}
+
+	// Retrieve environment variables
+	resourceInfo := os.Getenv("AKS_RESOURCE_ID")
+	if resourceInfo == "" {
+		customProperties["ACSResourceName"] = os.Getenv(envAcsResource)
+		customProperties["ClusterType"] = acsClusterType
+		customProperties["SubscriptionID"] = ""
+		customProperties["ResourceGroupName"] = ""
+		customProperties["ClusterName"] = ""
+		customProperties["Region"] = ""
+	} else {
+		customProperties["AKS_RESOURCE_ID"] = resourceInfo
+		splitStrings := strings.Split(resourceInfo, "/")
+		subscriptionID := ""
+		resourceGroupName := ""
+		clusterName := ""
+		if len(splitStrings) >= 9 {
+			subscriptionID = splitStrings[2]
+			resourceGroupName = splitStrings[4]
+			clusterName = splitStrings[8]
+		} else {
+			aiLogger.Fatalf("Error parsing AKS_RESOURCE_ID: %s", resourceInfo)
+		}
+		customProperties["ClusterType"] = aksClusterType
+		customProperties["SubscriptionID"] = subscriptionID
+		customProperties["ResourceGroupName"] = resourceGroupName
+		customProperties["ClusterName"] = clusterName
+		customProperties["Region"] = os.Getenv(envAksRegion)
```

Comment:
This code needs to be updated as per the changes in this PR - https://github.com/microsoft/Docker-Provider/commit/7559a07b5d28d8e223a665945fe139a53359d109

Reply:
I think I already have updated this. Do you see anything missing?